File: 15780.py
N, K = map(int, input().split())
a = list(map(int, input().split()))

tap = 0
for i in a:
    if i % 2:
        tap += (i // 2 + 1)
    else:
        tap += (i // 2)

if N <= tap:
    print("YES")
else:
    print("NO")

File: 15780.cpp
#include <iostream>
using namespace std;

int main() {
	int N, K; cin >> N >> K;

	int tap = 0;
	for (int i = 0; i < K; i++) {
		int a; cin >> a;

		if (a % 2) tap += (a / 2 + 1);
		else tap += (a / 2);
	}

	if (N <= tap) cout << "YES\n";
	else cout << "NO\n";
}
